worker: compile failure on non-linux

## Summary

Since #5543 got merged, `worker/workercmd` (and therefore `cmd/concourse`) no longer compiles on non-linux platforms.

## Steps to reproduce

1. `go build ./cmd/concourse` on a non-linux platform (say, darwin in my case)

## Expected results

Successful build with `concourse` binary output.

## Actual results

```bash
# github.com/concourse/concourse/worker/workercmd
worker/workercmd/guardian.go:32:15: cmd.Garden.Config undefined (type GardenBackend has no field or method Config)
worker/workercmd/guardian.go:33:53: cmd.Garden.Config undefined (type GardenBackend has no field or method Config)
worker/workercmd/guardian.go:50:38: cmd.Garden.DNSServers undefined (type GardenBackend has no field or method DNSServers)
worker/workercmd/guardian.go:60:15: cmd.Garden.DNS undefined (type GardenBackend has no field or method DNS)
worker/workercmd/guardian.go:61:29: cmd.dnsProxyRunner undefined (type *WorkerCommand has no field or method dnsProxyRunner)
worker/workercmd/guardian.go:89:15: cmd.Garden.Bin undefined (type GardenBackend has no field or method Bin)
worker/workercmd/guardian.go:90:19: cmd.Garden.Bin undefined (type GardenBackend has no field or method Bin)
worker/workercmd/guardian.go:97:3: unknown field 'Pdeathsig' in struct literal of type syscall.SysProcAttr
```


## Additional context

Failures on on [windows](https://ci.concourse-ci.org/teams/main/pipelines/concourse/jobs/build-rc/builds/202#L5ec5d2e6:2:10) and [darwin](https://ci.concourse-ci.org/teams/main/pipelines/concourse/jobs/build-rc/builds/202#L5ec5d2e7:24:32) on CI.

## Triaging info
 
* Concourse version: v6.2.0-pre (master)
* Browser (if applicable): n/a
* Did this used to work? yes

--- a/worker/workercmd/guardian.go
+++ b/worker/workercmd/guardian.go
@@ -1,17 +1,20 @@
+// +build linux
+
 package workercmd
 
 import (
-	"code.cloudfoundry.org/lager"
-	"code.cloudfoundry.org/localip"
 	"fmt"
-	concourseCmd "github.com/concourse/concourse/cmd"
-	"github.com/tedsuo/ifrit"
-	"github.com/tedsuo/ifrit/grouper"
 	"os"
 	"os/exec"
 	"path/filepath"
 	"strings"
 	"syscall"
+
+	"code.cloudfoundry.org/lager"
+	"code.cloudfoundry.org/localip"
+	concourseCmd "github.com/concourse/concourse/cmd"
+	"github.com/tedsuo/ifrit"
+	"github.com/tedsuo/ifrit/grouper"
 )
 
 // This runs Guardian using the gdn binary

